test/helpers/helpers.go

```diff
@@ -96,6 +96,33 @@ func WaitForSuccessfulDeployment(t *testing.T, deployment apps.Deployment) error
 	return nil
 }
 
+// WaitForNotFound polls the cluster for a particular resource name and namespace.
+// If the request fails because of the resource is found, it retries until the specified timeout.
+// If it succeeds in receiving an IsNotFound error, it returns nil.
+func WaitForNotFound(t *testing.T, result runtime.Object, namespace, name string) error {
```

Comment:
This function exists in PRs opened by both @kevinrizza and myself - the race is on to see who won't have to rebase.